parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -405,6 +405,30 @@ ensureDHCPv6() {
 ensureKubelet() {
     KUBELET_DEFAULT_FILE=/etc/default/kubelet
     mkdir -p /etc/default
+
+    # In k8s >= 1.29 kubelet no longer sets node internalIP when using external cloud provider
+    # https://github.com/kubernetes/kubernetes/pull/121028
+    # This regresses node startup performance in Azure CNI Overlay and Podsubnet clusters, which require the node to be
+    # assigned an internal IP before configuring pod networking.
+    # To improve node startup performance, explicitly set `--node-ip` to the IP returned from IMDS so kubelet sets
+    # the internal IP when it registers the node.
+    # If this fails, skip setting --node-ip, which is safe because cloud-node-manager will assign it later anyway.
+    if semverCompare ${KUBERNETES_VERSION:-"0.0.0"} "1.29.0"; then
```

Comment:
could we extract this to a separate function and wrap it in `logs_to_events`?

the event could be called something like `AKS.CSE.ensureKubelet.setNodeIPFlag`

Reply:
updated to move it into a helper function with `logs_to_events`. Can see that it's working in the provisioning logs:
<img width="996" alt="image" src="https://github.com/user-attachments/assets/2847c031-3493-46b9-bbb7-755c6cfc2862">